src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs

```diff
@@ -0,0 +1,160 @@
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class JsonConsoleFormatter : ConsoleFormatter, IDisposable
+    {
+        private IDisposable _optionsReloadToken;
+
+        public JsonConsoleFormatter(IOptionsMonitor<JsonConsoleFormatterOptions> options)
+            : base (ConsoleFormatterNames.Json)
+        {
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            string message = logEntry.Formatter(logEntry.State, logEntry.Exception);
+            if (logEntry.Exception == null && message == null)
+            {
+                return;
+            }
+            LogLevel logLevel = logEntry.LogLevel;
+            string category = logEntry.Category;
+            int eventId = logEntry.EventId.Id;
+            Exception exception = logEntry.Exception;
+            const int DefaultBufferSize = 1024;
+            using (var output = new PooledByteBufferWriter(DefaultBufferSize))
+            {
+                using (var writer = new Utf8JsonWriter(output, FormatterOptions.JsonWriterOptions))
+                {
+                    writer.WriteStartObject();
+                    string timestamp = null;
+                    var timestampFormat = FormatterOptions.TimestampFormat;
+                    if (timestampFormat != null)
+                    {
+                        var dateTime = FormatterOptions.UseUtcTimestamp ? DateTime.UtcNow : DateTime.Now;
+                        timestamp = dateTime.ToString(timestampFormat);
+                    }
+                    writer.WriteString("Timestamp", timestamp);
+                    writer.WriteNumber(nameof(logEntry.EventId), eventId);
+                    writer.WriteString(nameof(logEntry.LogLevel), GetLogLevelString(logLevel));
+                    writer.WriteString(nameof(logEntry.Category), category);
+                    writer.WriteString("Message", message);
+
+                    if (exception != null)
+                    {
+                        writer.WriteStartObject(nameof(Exception));
+                        writer.WriteString(nameof(exception.Message), exception.Message.ToString());
+                        writer.WriteString("Type", exception.GetType().ToString());
+                        writer.WriteStartArray(nameof(exception.StackTrace));
+                        string stackTrace = exception?.StackTrace;
+                        if (stackTrace != null)
+                        {
+#if NETSTANDARD2_0
+                            foreach (var stackTraceLines in stackTrace?.Split(new string[] { Environment.NewLine }, StringSplitOptions.None))
+#else
+                            foreach (var stackTraceLines in stackTrace?.Split(Environment.NewLine))
```

Comment:
/cc: @ericstj @davidfowl 
FYI, on what we have for stacktrace in Json formatter.
Please let me know if the feedback in this conversation thread is not yet addressed: https://github.com/dotnet/runtime/pull/38616#discussion_r453118538%20%20